src/BSeq.jl

```diff
@@ -0,0 +1,209 @@
+"""
+Solve Bethe-Slapter equation
+"""
+
+module BSeq
+
+using ..Parameter, ..Convention, ..LegendreInteraction
+using ..Parameters, ..GreenFunc, ..Lehmann, ..CompositeGrids
+
+const freq_sep = 0.01
+
+function initFR(Euv, rtol, sgrid, param)
+    @unpack β, me, μ = param
+    Ω_c = freq_sep
+
+    wn_mesh = GreenFunc.ImFreq(β, FERMION; Euv=Euv, rtol=rtol, symmetry=:pha)
+    R_freq = GreenFunc.MeshArray(wn_mesh, sgrid; dtype=Float64)
+    F_freq = similar(R_freq)
+    for ind in eachindex(R_freq)
+        ni, ki = ind[1], ind[2]
+        ωn, k = matfreq(wn_mesh)[ni], sgrid[ki]
+        e = k^2 / 2 / me - μ
+        R_freq[ni, ki] = (1.0 - 2.0 * ωn^2 / (ωn^2 + Ω_c^2)) / (Ω_c^2 + e^2)
+    end
+    R_imt = real(R_freq |> to_dlr |> to_imtime)
+    R_ins = GreenFunc.MeshArray([1], sgrid; dtype=Float64, data=zeros(1, sgrid.size))
+    return F_freq, R_imt, R_ins
+end
+
+function calcF!(F::GreenFunc.MeshArray, R::GreenFunc.MeshArray, R_ins::GreenFunc.MeshArray, G2::GreenFunc.MeshArray)
+    R_freq = R |> to_dlr |> to_imfreq
+    for ind in eachindex(F)
+        F[ind] = real(R_freq[ind] + R_ins[1, ind[2]]) * G2[ind]
+    end
+end
+
+function calcR!(F::GreenFunc.MeshArray, R::GreenFunc.MeshArray, R_ins::GreenFunc.MeshArray, kernel, kernel_bare, qgrids)
+    kgrid = F.mesh[2]
+    F_dlr = F |> to_dlr
+    F_imt = real(F_dlr |> to_imtime)
+    F_ins = real(dlr_to_imtime(F_dlr, [F.mesh[1].representation.β,])) * (-1)
+    for ind in eachindex(R)
+        τi, ki = ind[1], ind[2]
+        Fq = CompositeGrids.Interp.interp1DGrid(view(F_imt, τi, :), kgrid, qgrids[ki].grid)
+        integrand = view(kernel, ki, 1:qgrids[ki].size, τi) .* Fq
+        R[τi, ki] = CompositeGrids.Interp.integrate1D(integrand, qgrids[ki]) ./ (-4 * π * π)
+        if τi == 1
+            Fq = CompositeGrids.Interp.interp1DGrid(view(F_ins, 1, :), kgrid, qgrids[ki].grid)
+            integrand = view(kernel_bare, ki, 1:qgrids[ki].size) .* Fq
+            R_ins[1, ki] = CompositeGrids.Interp.integrate1D(integrand, qgrids[ki]) ./ (-4 * π * π)
+        end
+    end
+end
+
+function calcR_2d!(F::GreenFunc.MeshArray, R::GreenFunc.MeshArray, R_ins::GreenFunc.MeshArray, kernel, kernel_bare, qgrids)
+    kgrid = F.mesh[2]
+    F_dlr = F |> to_dlr
+    F_imt = real(F_dlr |> to_imtime)
+    F_ins = real(dlr_to_imtime(F_dlr, [F.mesh[1].representation.β,])) * (-1)
+    for ind in eachindex(R)
```

Comment:
for (\taui, ki) in eachindex(R)

Reply:
ERROR: iteration is deliberately unsupported for CartesianIndex. Use `I` rather than `I...`, or use `Tuple(I)...`